### PR TITLE
[AIRFLOW-3746] Fix DockerOperator missing container exit

### PR DESCRIPTION
**Pull request by ashwiniadiga**
My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-3746)

switch from cli.logs to cli.attach to prevent missing container exit

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [(https://issues.apache.org/jira/browse/AIRFLOW-3746)] 
  - https://issues.apache.org/jira/browse/AIRFLOW-3746

### Description

- My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) 
switch from cli.logs to cli.attach to prevent missing container exit
With DockerOperator and using the celery executor, the task runs but never completes. It hungs as 
container.logs / certain container hangs.

since The contianer.logs() function is a wrapper around container .attach() method, which you can use instead if you want to fetch/stream container output without first retrieving the entire backlog.
So changing the container.logs() to container.attach() gives the same output and completes the task.
with the following log.
The docker_operator Task exited with return code 0

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:All the existing tests are fixed. No new tests are not required.

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/operators/docker_operator.py [hunk 1]
```diff
@@ -228,7 +228,10 @@ def execute(self, context):
             self.cli.start(self.container['Id'])
 
             line = ''
-            for line in self.cli.logs(container=self.container['Id'], stream=True):
+            for line in self.cli.attach(container=self.container['Id'],
+                                        stdout=True,
+                                        stderr=True,
+                                        stream=True):
```
Comment (zhongjiajie) [inline airflow/operators/docker_operator.py:237]: You have to create JIRA ticket,because you change code

**Review by zhongjiajie — COMMENTED** [review]
Create JIRA tickets, please

File: airflow/operators/docker_operator.py (hunk 1 above)
Comment (zhongjiajie) [reply airflow/operators/docker_operator.py:237]: @ashwiniadiga You CI test failed due to you don not change unit test code, please change `tests/operators/test_docker_operator.py` to pass CI test

Comment (ashwiniadiga) [reply airflow/operators/docker_operator.py:237]: https://issues.apache.org/jira/browse/AIRFLOW-3746, 
I will fix the unit tests

Comment (ashwiniadiga) [reply airflow/operators/docker_operator.py:237]: @zhongjiajie I have fixed the  all failing tests. Can you please review

Comment (zhongjiajie) [reply airflow/operators/docker_operator.py:237]: @ashwiniadiga LGTM

**Comment by zhongjiajie** [comment]
@ashb @Fokko @feng-tao PLAL, @ashwiniadiga send a dev-mail for asking committer review and merge to master.

**Comment by ashb** [comment]
The PR template has a section for description of what this PR does - please fill it in as I can't immediately tell why this change is needed

**Review by ashb — CHANGES_REQUESTED** [review]
Explain why this is needed please.

**Comment by ashwiniadiga** [comment]
Done

**Comment by ashwiniadiga** [comment]
Done

File: tests/operators/test_docker_operator.py [hunk 2]
```diff
@@ -49,6 +49,7 @@ def test_execute(self, client_class_mock, mkdtemp_mock):
         client_mock.create_container.return_value = {'Id': 'some_id'}
         client_mock.create_host_config.return_value = host_config
         client_mock.images.return_value = []
+        client_mock.attach.return_value = ['container log']
         client_mock.logs.return_value = ['container log']
```
Comment (ashb) [inline tests/operators/test_docker_operator.py:46]: Probably don't need this mock anymore?

File: tests/operators/test_docker_operator.py [hunk 3]
```diff
@@ -95,6 +97,7 @@ def test_execute_tls(self, client_class_mock, tls_class_mock):
         client_mock.create_container.return_value = {'Id': 'some_id'}
         client_mock.create_host_config.return_value = mock.Mock()
         client_mock.images.return_value = []
+        client_mock.attach.return_value = []
         client_mock.logs.return_value = []
```
Comment (ashb) [inline tests/operators/test_docker_operator.py]:
```suggestion
```

File: tests/operators/test_docker_operator.py [hunk 4]
```diff
@@ -121,6 +124,7 @@ def test_execute_unicode_logs(self, client_class_mock):
         client_mock.create_container.return_value = {'Id': 'some_id'}
         client_mock.create_host_config.return_value = mock.Mock()
         client_mock.images.return_value = []
+        client_mock.attach.return_value = ['unicode container log 😁']
         client_mock.logs.return_value = ['unicode container log 😁']
```
Comment (ashb) [inline tests/operators/test_docker_operator.py]:
```suggestion
```

File: tests/operators/test_docker_operator.py [hunk 5]
```diff
@@ -143,6 +147,7 @@ def test_execute_container_fails(self, client_class_mock):
         client_mock.create_container.return_value = {'Id': 'some_id'}
         client_mock.create_host_config.return_value = mock.Mock()
         client_mock.images.return_value = []
+        client_mock.attach.return_value = []
         client_mock.logs.return_value = []
```
Comment (ashb) [inline tests/operators/test_docker_operator.py]:
```suggestion
```

File: tests/operators/test_docker_operator.py [hunk 6]
```diff
@@ -172,6 +177,7 @@ def test_execute_no_docker_conn_id_no_hook(self, operator_client_mock):
         client_mock = mock.Mock(name='DockerOperator.APIClient mock', spec=APIClient)
         client_mock.images.return_value = []
         client_mock.create_container.return_value = {'Id': 'some_id'}
+        client_mock.attach.return_value = []
         client_mock.logs.return_value = []
```
Comment (ashb) [inline tests/operators/test_docker_operator.py]:
```suggestion
```

File: tests/operators/test_docker_operator.py [hunk 7]
```diff
@@ -207,6 +213,7 @@ def test_execute_with_docker_conn_id_use_hook(self, operator_client_mock,
...
```
Comment (ashb) [inline tests/operators/test_docker_operator.py]:
```suggestion
```

**Review by ashb — CHANGES_REQUESTED** [review]
I think we don't need to mock `client.logs` anymore?

File: tests/operators/test_docker_operator.py (hunk 2 above)
Comment (ashwiniadiga) [reply tests/operators/test_docker_operator.py:46]: When self.xcom_push_flag is true, its still getting it from cli.logs. So keeping the this mock

**Review by ashb — APPROVED** [review]

**Review by Fokko — APPROVED** [review]

**Comment by ashb** [comment]
@ashwiniadiga Could you rebase on to master and see if this fixes the build?

**Pull request merged**